File: java-grpc-spring-boot-module/grpc-server/src/main/java/com/binghe/member/MemberGrpcServiceImpl.java
package com.binghe.member;

import com.binghe.proto.GetMemberRequest;
import com.binghe.proto.GetMemberResponse;
import com.binghe.proto.MemberGrpcServiceGrpc;
import io.grpc.stub.StreamObserver;
import lombok.RequiredArgsConstructor;
import net.devh.boot.grpc.server.service.GrpcService;

@RequiredArgsConstructor
@GrpcService
public class MemberGrpcServiceImpl extends MemberGrpcServiceGrpc.MemberGrpcServiceImplBase {

    private final MemberService memberService;

    @Override
    public void getMember(GetMemberRequest request, StreamObserver<GetMemberResponse> responseObserver) {
        Long id = (long) request.getId();
        Member member = memberService.findById(id);
        GetMemberResponse response = GetMemberResponse.newBuilder()
                .setId(Math.toIntExact(member.getId()))
                .setName(member.getName())
                .setAge(member.getAge())
                .setDescription(member.getDescription())
                .build();
        responseObserver.onNext(response);
        responseObserver.onCompleted();
    }

    @Override
    public void getMemberServerStream(GetMemberRequest request, StreamObserver<GetMemberResponse> responseObserver) {
        super.getMemberServerStream(request, responseObserver);
    }

    @Override
    public StreamObserver<GetMemberRequest> getMemberClientStream(StreamObserver<GetMemberResponse> responseObserver) {
        return super.getMemberClientStream(responseObserver);
    }

    @Override
    public StreamObserver<GetMemberRequest> getMemeberBiStream(StreamObserver<GetMemberResponse> responseObserver) {
        return super.getMemeberBiStream(responseObserver);
    }
}

File: kotlin-grpc-coroutine-spring-boot/grpc-server/src/main/kotlin/com/binghe/member/MemberGrpcServiceImpl.kt
package com.binghe.member

import com.binghe.MemberService
import com.binghe.proto.GetMemberRequest
import com.binghe.proto.GetMemberResponse
import com.binghe.proto.MemberGrpcServiceGrpcKt
import org.lognet.springboot.grpc.GRpcService

@GRpcService
class MemberGrpcServiceImpl(private val memberService: MemberService) : MemberGrpcServiceGrpcKt.MemberGrpcServiceCoroutineImplBase() {

    override suspend fun getMember(request: GetMemberRequest): GetMemberResponse {
        val member = memberService.findById(request.id)
        return GetMemberResponse.newBuilder()
            .setId(member?.id!!)
            .setName(member.name)
            .setAge(member.age ?: 0)
            .setDescription(member.description)
            .build()
    }
}